### PR TITLE
Update to Symfony 6, PHP-CS-Fixer 3 and PHPUnit 9.5

### LINKED ISSUE #123: Support Symfony 6
Hey!

Currently the `symfony/console` component is limited to `^5.0`.

Any plans to support Symfony 6?

### PR DESCRIPTION
**Pull request by Jubeki**
Closes #123 

### Changed:
- Updated to Symfony\Console `^6.0`
- Updated to PHP-CS-Fixer `^3.0`
- Updated to PHPUnit `^9.5`
    - Rename `assertContains` to `assertStringContainsString`
    - Rename `assertNotContains ` to `assertStringNotContainsString`
    - Rename `assertFileNotExists' to 'assertFileDoesNotExist'
- Updated to PHP `^8.0`
- Updated GitHub Actions workflows
- Make constants public (Hook.php)
- Cast `shell_exec` output to string (src/helpers.php) because `trim` required a `string` and not `string|ƒalse|null`

### Added
- Added `file_exists` check before create a directory

EDIT:
For these wanting to try out these changes you can do the following to your composer.json:
```json
{
    "repositories": [
        {
            "type": "git",
            "url": "https://github.com/Jubeki/composer-git-hooks"
        }
    ],
    "require-dev": {
        "brainmaestro/composer-git-hooks": "dev-prepare-for-symfony-6"
    },
    "minimum-stability": "dev",
    "prefer-stable": true
}
```

### CODE REVIEW

File: composer.json
Comment (NicolasGuilloux) [inline composer.json:20]: This makes it incompatible with any other version, and there is apparently no reason to lock it. The same for the PHP version. It would be better to require the following versions :

```
"require" : {
    "php": "^7.0 || ^8.0",
    "symfony/console": "^3.0 || 4.0 || 5.0 || 6.0"
}
```

It should be obviously tested against any breaking change between versions.

Comment (NicolasGuilloux) [reply composer.json:20]: I partially agree about that from my point of view.

- About the outdated version, you're completely right about migrating. It is really bad to have outdated Symfony or PHP running...

- ...but it shouldn't be the maintainer responsibility to force people update their library. If a specific feature is required by a more recent PHP version or symfony console, I agree that the maintener shouldn't bother maintaining for older versions. But if it works, there is no need to lock it. I (very) quickly went through the code, and I don't see anything that justify locking the version.

- About the PHP-CS-Fixer, it is used only for development purpose. I don't agree that it should lock the version in production. The only reason it can block it would be for the CI, that is why PHP CS Fixer in its Readme advice to install it in the `tools/php-cs-fixer` folder to avoid conflicting with the environment.

- But everything is fine if old versions are still accessible. We can consider that outdated installation won't evolve and therefore, will not need any future fix provided by future updates. 

Thanks for the conversation though, I'm always glad to talk about that :)

Comment (Jubeki) [reply composer.json:20]: Yeah you are right with the PHP-CS-Fixer. This should only run in the CI or during development.
I also agree with the locking of the versions. Just to say that the current version is already only support Symfony 5 so it might be possible to at least support that version.

In the end the decision lies with @BrainMaestro so he can decide what he wants to support.

**Comment by rcerljenko** [comment]
@BrainMaestro any plans for merging this?

**Comment by jdecode** [comment]
Can't use this with Laravel 9, until this is approved/merged.

**Comment by Jubeki** [comment]
@jdecode you can use the workaround I provided in the Pull Request Description.

**Comment by jdecode** [comment]
@Jubeki  - Yes, I am using the workaround for now.
Much thanks for that actually!

And still waiting for PR merge, so that I can upgrade some projects to L9 (without the workaround) :+1:

**Comment by sirolad** [comment]
I am also waiting for this

**Comment by eduardoturconi** [comment]
I am also waiting for this

**Comment by SOSTheBlack** [comment]
+1

**Comment by AnjaLiebermann** [comment]
Can anyone of us do something to speed this up?

**Comment by rdgout** [comment]
Maybe it's about time to create a fork and move to that. @BrainMaestro doesn't seem to have time to merge a PR that's been opened since january.

File: composer.json
Comment (BrainMaestro) [reply composer.json:20]: I believe it's fine to update the versions. Users on EOL'd versions of PHP or Symfony Console should use older versions of the package

**Review by BrainMaestro — APPROVED** [review]
I appreciate the contribution. Sorry for the long wait

**Comment by BrainMaestro** [comment]
I created a new tag [v3.0.0-alpha.1](https://github.com/BrainMaestro/composer-git-hooks/releases/tag/v3.0.0-alpha.1). Check it out and let me know if there are any issues

**Comment by RaederDev** [comment]
Could you publish a stable, non alpha tag for this?

**Comment by TheFox** [comment]
I am also waiting for this.

**Comment by BrainMaestro** [comment]
Sorry for the long wait 😅, but there v3.0.0 now, and I'm able to devote more time to support this project, so it shouldn't take that long between releases anymore

**Pull request merged**
